Add local-run role assumption usage docs (#3851)
Yelp/paasta: docs/source/adhoc_instances.rst

    interactive:
      deploy_group: prod.canary
      mem: 10000

Assuming service role from another AWS account
----------------------------------------------

> If you need to run your instance with another account than your current environment's, e.g. use prod account from dev environment, you can specify ``--cluster`` and ``--assume-role-aws-account``.
> 
> In the example below, it's enough to specify ``--cluster`` so ``local-run`` will use configuration that are for ``pnw-prod`` and use prod account, no matter your current environment.

Reviewer: optional wording suggestions:
```suggestion
If you need to locally run your instance using a role from an AWS account that differs from your current environment's (e.g. you want to use a role from our production account in one of our dev environments), you will need to specify ``--cluster`` and ``--assume-role-aws-account``.

In the example below, specifying ``--cluster`` in the ``local-run`` command will use instance configurations for ``pnw-prod`` and ``--assume-pod-identity`` will use the configured role from the prod account, no matter your current environment.
```

.. code-block::sh

>     paasta local-run -s <service-name> --pull --assume-pod-identity -i <service-instance> --cluster pnw-prod -I

Reviewer: optional nit: imo, in documentation, we should try to use the long-form of options so that it's easier for folks to search for in the docs/in paasta --help
```suggestion
    paasta local-run --service <service-name> --pull --assume-pod-identity --instance <service-instance> --cluster pnw-prod --interactive
```

> Here, ``-I`` flag is used to get the interactive shell.

Reviewer: ```suggestion
Here, ``--interactive`` flag is used to get the interactive shell.
```

whoops, we missed one flag update :)
